### PR TITLE
Improvements for multi-layer fee UX

### LINKED ISSUE #13541: "Send max" doesn't work on Optimism
### Describe the bug

`Send max` results in a failed transaction on Optimism

### Steps to reproduce

1. Switch networks to Optimism
2. Make sure you have some amount of ETH
3. Click `send`
4. Fill in an address and click `MAX`
5. Submit the transaction
6. Notice that it fails

### Error messages or log output

_No response_

### Version

10.8.1

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### LINKED ISSUE #13542: Show currency estimate alongside ETH on Optimism
We'd ideally be able to show user's selected currency conversion alongside ETH on Optimism (like it appears on Mainnet). Right now, most users paste ETH values into a currency converter to understand the cost of their transaction. This is likely generalizable to other L2s as well. 

**Optimism:**
Screen Shot 2022-02-07 at 6 29 01 PM

**ETH mainnet:** 

### Steps to reproduce

1. Switch network to Optimism
2. Send ETH to someone
3. View the transaction screen ^

### Version

10.8.1

### Browser

Chrome

### Operating system

MacOS

### LINKED ISSUE #13543: Combine L1+L2 gas when using Optimism
### Describe the bug

Today, MM shows two gas estimates for all tx on Optimism, which causes confusion among users. We should instead show one total. This is possibly generalizable to other L2s

See docs [here](https://community.optimism.io/docs/guides/wallet-dev/#displaying-fees) for more detail.  

Screen Shot 2022-02-07 at 6 29 01 PM

### Steps to reproduce

1. Switch to Optimism
2. Submit a tx – notice both gas fees

### Error messages or log output

_No response_

### Version

10.8.1

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### LINKED ISSUE #13544: Don't allow users to set gas price on Optimism
### Describe the bug

User's shouldn't be able to edit their gas price or fee when using Optimism:

See docs [here](https://community.optimism.io/docs/guides/wallet-dev/#estimating-total-fees) for more detail. 
> Unlike with the standard execution gas fee, users cannot specify a particular gas price or gas limit for this portion of their transaction cost. Instead, this fee is automatically deducted from the user's ETH balance on Optimism when the transaction is executed.

### Steps to reproduce

1. Switch to Optimism
2. Initiate a new transaction
3. Notice the gas customization UI
4. Fiddle with it; see your tx likely fail

### Error messages or log output

_No response_

### Version

10.8.1

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by danjm**
Fixes #13541
Fixes #13542
Fixes #13543
Fixes #13544

https://user-images.githubusercontent.com/7499938/152933848-2f8af31e-8d14-44ee-a7e6-a57de4476677.mp4

### CODE REVIEW

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 1]
```diff
@@ -488,48 +484,44 @@ export default class ConfirmTransactionBase extends Component {
...
                 value={hexMinimumTransactionFee}
-                hideLabel={!useNativeCurrencyAsPrimaryCurrency}
-                numberOfDecimals={isMultiLayerFeeNetwork ? 18 : 6}
+                hideLabel={false && !useNativeCurrencyAsPrimaryCurrency}
+                numberOfDecimals={6}
               />
             </div>
           }
           subText={
-            !isMultiLayerFeeNetwork && (
```
Comment (danjm) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:514]: No code changes here except the removal of `!isMultiLayerFeeNetwork && ` and then all the code having to change indentation

**Comment by darkwing** [comment]
We need to run the `yarn verify-scripts:fix` command but this looks and works great!

**Comment by bdresser** [comment]
@danjm @darkwing thank you for putting this PR together! Anything we can help with around testing / QA to help get it over the finish line?

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 2]
```diff
@@ -492,48 +488,44 @@ export default class ConfirmTransactionBase extends Component {
...
           }
           detailTotal={
             <div className="confirm-page-container-content__currency-container">
               {renderHeartBeatIfNotInTest()}
               <UserPreferencedCurrencyDisplay
                 type={PRIMARY}
                 value={hexMinimumTransactionFee}
-                hideLabel={!useNativeCurrencyAsPrimaryCurrency}
-                numberOfDecimals={isMultiLayerFeeNetwork ? 18 : 6}
+                hideLabel={false && !useNativeCurrencyAsPrimaryCurrency}
```
Comment (darkwing) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:506]: Looks like leftover debug?

**Review by darkwing — CHANGES_REQUESTED** [review]

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js (hunk 2 above)
Comment (danjm) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:506]: yes, nice catch

**Review by darkwing — APPROVED** [review]

**Comment by adonesky1** [comment]
Also is there some reason we don't show the **Total L2 Gas Fees** on the transaction details view?

File: ui/ducks/send/send.js [hunk 3]
```diff
@@ -313,12 +318,14 @@ async function estimateGasLimitForSend({
       error.message.includes('Transaction execution error.') ||
       error.message.includes(
         'gas required exceeds allowance or always failing transaction',
-      );
+      ) ||
+      (CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP[chainId] &&
+        error.message.includes('gas required exceeds allowance'));
```
Comment (adonesky1) [inline ui/ducks/send/send.js:323]: Is this redundant? Or are `gas required exceeds allowance` and `gas required exceeds allowance or always failing transaction` separate and different error messages?

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 4]
```diff
@@ -491,16 +487,14 @@ export default class ConfirmTransactionBase extends Component {
...
-              <div className="confirm-page-container-content__currency-container">
-                {renderHeartBeatIfNotInTest()}
-                <UserPreferencedCurrencyDisplay
-                  type={SECONDARY}
-                  value={hexMinimumTransactionFee}
-                  hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
-                />
-              </div>
-            )
+            <div className="confirm-page-container-content__currency-container test">
```
Comment (adonesky1) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:490]: is `test` supposed stay there in the className?

**Review by adonesky1 — COMMENTED** [review]
A few questions/comments

**Review by brad-decker — APPROVED** [review]

**Pull request merged**